### ISSUE TITLE
S3900 FP: `is` is not considered as null-check

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2775]
### Description

[from community](https://community.sonarsource.com/t/s3900-fp-on-c-arguments-of-public-methods-should-be-validated-against-null/15858)

Validation using `is` is not taken into consideration when chaining bool expressions.

### Repro steps

```
    public class ReproWithIs
    {
        public override bool Equals(object obj)
        {
            var equals = (obj is string) && (obj.GetHashCode() == GetHashCode()); // Noncompliant FP
            if (equals)
            {
                // do stuff
            }
            return equals;
        }
    }
```

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* SonarQube 8.0
* SonarC# 8.0
* SonarScanner for MSBuild 4.7.1
* MSBuild 15.9

**Comment by costin-zaharia-sonarsource** [comment]
I've found this also on peach: https://peach.sonarsource.com/project/issues?id=servuo&issues=AWiU6Wnb02kLojS0BHMT&open=AWiU6Wnb02kLojS0BHMT

**Comment by pavel-mikula-sonarsource** [comment]
Fixed by #6793
